### ISSUE TITLE
Situation, Directions, and Outside/Garden extras request

### ISSUE THREAD

**Issue by adam-hf** [reapit/foundations#7541]
Hello, is it possible to make the data from the extra fields Situation, Directions, and Outside/Garden (from the Marketing Details > Extra Descriptions section of AgencyCloud) present when requesting property data from the Foundations REST API for the customer GTH?

Additionally, via the older web-services API we used to receive data in fields named Mileage and Location, are these fields also present for GTH as extra fields that need configuration to appear in Foundations? Are any of these extra fields (Long Description, Situation, Directions, and Outside Garden etc.) common across agencies, or does each agency with extra fields have them named differently when output by the Foundations REST API?

In summary:

- The name of the customer is Greenslade Taylor Hunt (Reapit customer ID: GTH).
- The name of the fields we need surfaced in the Foundations REST API are: Situation, Directions, and Outside/Garden.
- A screenshot of the fields in the AgencyCloud user interface is attached.
- We require the data as this is where the client stores some property description text, which we need to fetch from the Foundations REST API to display on the individual property listing page of the customer's website.

[Screenshots Removed as they contain PII]

**Comment by quigger** [comment]
Thanks @plittlewood-rpt 

Is this really going to be 1-4 months to complete? I would have thought these extra data fields would already be scoped in the Foundations API as they are existing fields in WebServices.

We can't get a full property description without this data, I would have thought other Foundation users would have come across this already.

Can it be expediated please?

**Comment by HollyJoyPhillips** [comment]
Hi @quigger, this issue is in our current sprint. Tickets are initially added to 'Near term' until they are assigned to a sprint.

**Comment by quigger** [comment]
Hi @HollyJoyPhillips thanks.

The comment above says:
"This issue has been updated and moved to our ‘Near Term’ column (typically completed within 1 - 4 months). "

When can I expect that to be competed then please?

**Comment by HollyJoyPhillips** [comment]
Hi @quigger Odd, you should have had another notification when it was assigned to a sprint: 

"plittlewood-rpt added this to To do in [Back end sprint 75]"

I'll look into that. 

We work in weekly sprints so I'd like to say, if there are no issues or other dependencies, we should aim for the development work to be completed this week. 

You can find out more about our processes here: https://foundations-documentation.reapit.cloud/dev-requests

**Comment by quigger** [comment]
Hi @HollyJoyPhillips 

I saw it was in [Back end sprint 75] but that has not time period for me, so I had no idea when that would occur.

Thanks for this:
"We work in weekly sprints so I'd like to say, if there are no issues or other dependencies, we should aim for the development work to be completed this week."

**Comment by plittlewood-rpt** [comment]
Hi @quigger access for the fields mentioned above has been set up and should become active within the next 15 minutes.

Field names:

Situation -> pSituation
Directions -> pDirections
Outside/Garden -> pOutGar

As per any other semi structured data you access it using the `extrasField` query string parameter eg

`GET /properties?extrasField=pSituation&extrasField=pDirections`

or

`GET /properties/{id}?extrasField=pSituation&extrasField=pDirections`

In response to the rest of your query, there is sometimes some commonality between field names in other customer systems but it's best never to assume that this will be the case.

I'm not sure what your mileage fields relate to in the old web services, do you have any further information? Mileage would suggest the distance between two points so it feels like there's some additional context somewhere.

Any issues please let us know.

**Comment by quigger** [comment]
Hi @plittlewood-rpt 

Thanks.

> access for the fields mentioned above has been set up and should become active within the next 15 minutes.

Super.

> In response to the rest of your query, there is sometimes some commonality between field names in other customer systems but it's best never to assume that this will be the case.

But we have no way to check right? So if we can't see a field - we will have to ask you if it exists and request the field to be enabled?

And if the client is NEW to Homeflow (e.g. GTH) - we do not know the old fields?

> I'm not sure what your mileage fields relate to in the old web services, do you have any further information? Mileage would suggest the distance between two points so it feels like there's some additional context somewhere.

It was an extra 'description' type field about where the property was I think.

I found a client with that field: Marchand Petit.
See: https://tracker.reapit.net/demo/_support/web-service/method/?wsdl=https%3A%2F%2Fwebservice.reapit.net%2Fmar%2F%3Fwsdl&soapmethod=GetGeneralProperty&id=homeflow&pwd=2ea241ad174d4d6

Example data:

ID: rps_mar-MOD220084
Mileage: A38 7 miles, Kingsbridge 7.5 miles, Modbury 3 miles, Ivybridge 6.5 miles (distances approximate)

ID: rps_mar-MOD220076
Mileage: A38 7 miles, Kingsbridge 7.5 miles, Modbury 3 miles, Ivybridge 6.5 miles (distances approximate)

ID: rps_mar-KIN180237
Mileage: Totnes 13 miles; A38 Devon Expressway 10 miles; Salcombe 6 miles (distances are approximate)

**Comment by plittlewood-rpt** [comment]
Thanks, I've checked and the Mileage field in the system above is just an additional description, nothing fancy. This does not exist in the GTH build, nor does a description called Location.

For new customers to Homeflow, you would need to understand their system. We deliberately do not advertise names of custom fields as that makes them open to misuse. The core Platform API models are designed to be a standard suite of APIs that serve all datasets, so a developer can build an app for one customer and know that it will work for other customers. The exception here is the `extrasField` functionality which must be explicitly enabled on a customer by customer basis as documented.

I hope this helps

**Comment by plittlewood-rpt** [comment]
@quigger I should have added, it may be necessary to rotate your API key now as you pasted the full link (including the key) as this is a public board

**Comment by adam-hf** [comment]
@plittlewood-rpt How would we go about getting a new API key for the SOAP web-service? I've had a look at the web-services documentation and I can't see any information on this.

**Comment by plittlewood-rpt** [comment]
I've raised this with our internal infra team. I'll report back once I've heard from them

**Comment by plittlewood-rpt** [comment]
@adam-hf our internal infrastructure team are looking at rotating this API key

**Comment by aalim405** [comment]
@adam-hf - we will be executing this now and will be sharing the new API Key and passwords through our normal process.

**Comment by aalim405** [comment]
@quigger 
1. SI Change the API Key and Password
2. SI Share this information internally to our Service Desk
3. Service Desk share this info using a secure method to Homeflow via the Jira ticket and email - can you confirm which email address we should be using to share info?

**Comment by quigger** [comment]
Hi @aalim405 

I would guess we ought to share with the main contact on Reapit Foundations which I think is bob@homeflow.co.uk

Thanks.

**Comment by quigger** [comment]
@aalim405 

Thanks - that's all updated.
